src/cleanfigure.m

```diff
@@ -410,11 +410,13 @@ function simplifyLine(meta, handle, targetResolution)
 
     % If the path has markers, perform pixelation instead of simplification
     hasMarkers = ~strcmpi(get(handle,'Marker'),'none');
-    if hasMarkers
+    hasLines   = ~strcmpi(get(handle,'LineStyle'),'none');
+    if hasMarkers && ~hasLines
         % Pixelate data at the zoom multiplier
         mask      = pixelate(xData, yData, xToPix, yToPix);
         id_remove = find(mask==0);
-    else
```

Comment:
I would go for 
```matlab
if hasMarkers && ~hasLines
    ...
elseif ~hasMarkers && hasLines
    ...
end
```

that makes it clearer that both are mutually exclusive (while independent `if` statements need not be).